web/src/modules/layout/hooks/useOnClickOutside.js

```diff
@@ -0,0 +1,20 @@
+// @flow
+
+import { useEffect } from 'react'
+
+const useOnClickOutside = (ref: {current: ?HTMLElement}, callback: () => void) => {
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (event.target instanceof Node && ref.current && !ref.current.contains(event.target)) {
+        callback()
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside)
```

Comment:
Is it possible here to watch for other event listeners as well?

See https://issues.tuerantuer.org/browse/IGAPP-10 and https://issues.tuerantuer.org/browse/IGAPP-29

Reply:
Done